omod/src/main/webapp/resources/scripts/jquery-ui/js/openmrsSearch.js

```diff
@@ -545,30 +547,23 @@ function OpenmrsSearch(div, showIncludeVoided, searchHandler, selectionHandler,
                 }
             });
 
-            // Browser back button support, for preserve data
-            if (lastSearchParams !== null) {
-                $j('#inputNode').val(lastSearchParams.lastSearchText);
-                if (lastSearchParams.includeVoided == 1) {
-                    $j('#includeVoided').attr('checked','checked');
-                }
-                if (lastSearchParams.includeVerbose == 1) {
-                    $j('#includeVerbose').attr('checked','checked');
+            //Browser back button support, to preserve data
+            if(history.state || lastSearchParams.lastSearchText){
+            	//check for lastSearchParams so we don't overwrite something sent as a option
+                //if it is not set check to see if there are parameters to get from history.pushState
+            	if(!lastSearchParams.lastSearchText){
```

Comment:
i would assume that searchPhrase, includeVoided and showVerbose should all be treated just the same, priority should be the widget options then history. You can also copy the history into the options just like the search phrase and get rid of lastSearchParams